packages/core/src/amazonqGumby/chat/controller/controller.ts

```diff
@@ -451,9 +460,18 @@ export class GumbyController {
             this.messenger.sendKnownErrorResponse('no-alternate-dependencies-found', message.tabID)
             await this.continueTransformationWithoutHIL(message)
         } else if (message.error instanceof ModuleUploadError) {
-            this.resetTransformationChatFlow()
+            this.messenger.sendUnrecoverableErrorResponse('upload-to-s3-failed', message.tabID)
```

Comment:
In my earlier PR (https://github.com/aws/aws-toolkit-vscode/pull/5132/files#r1633998488), I kept the `resetTransformationChatFlow()` but deleted the `sendUnrecoverableErrorResponse()` calls as they were causing duplicate chat messages to show. What is the behavior with this change?

Reply:
Oof, I see that now. I'll revert this behavior to match your implementation - the intention was that the `controller` and the `messenger` class controls what is displayed in the chat, it's not ideal to have that functionality scattered across multiple classes, but this is refactoring better done in a separate PR with testing